modules/hostdb/scan.go

```diff
@@ -79,10 +95,60 @@ func (hdb *HostDB) threadedProbeHost(entry *modules.HostEntry) {
 // every few hours to see who is online and available for uploading.
 func (hdb *HostDB) threadedScan() {
 	for {
+		// Determine who to scan. At most 'MaxActiveHosts' will be scanned,
+		// starting with the active hosts followed by a random selection of the
+		// inactive hosts.
+		id := hdb.mu.Lock()
+		{
+			// Check all of the active hosts.
+			for _, host := range hdb.activeHosts {
+				go hdb.threadedProbeHost(host.hostEntry)
+			}
```

Comment:
I did realize that some other process might remove the host from active hosts between the locks, so then threadedProbeHost might be acting on a missing host.

\>.< foo